File: Euler/7/7.c
#include <stdio.h>
#include <math.h>

int f(int n) {
 int limit = sqrt ( n );
 for (int i=3;i <= limit;i+=2)
  if (n%i==0) return 0;
 return 1;
}

int primepos(int n){
 if (n==1) { return 2; }
 int p, pn;
 p = 3; pn = 2;
 for (int i = p + 2; pn<n; i+=2)
  if ( f(i) ) { p=i;pn++; }
 return p;
}


int main() {
 printf("%i\n",primepos(10001));
 return 0;
}

File: Euler/7/7.lua
#!/usr/bin/lua5.3

-- Returns primeposition, primenumber
function primepos(n)
 local function f(n)
  for i=3,math.floor(math.sqrt(n)),2 do
   if n%i==0 then return false end
  end
  return true
 end

 local p,pn = 3, 2
 while pn<n do
  p = p + 2
  if f(p) then pn = pn+1 end
 end
 return pn,p
end

print( string.format("%i: %i",primepos(10001) ) )
